app/controllers/allocation_staff_controller.rb

```diff
@@ -7,15 +7,15 @@ class AllocationStaffController < PrisonsApplicationController
 
   def index
     @case_info = Offender.find_by!(nomis_offender_id: prisoner_id_from_url).case_information
-    @allocation = AllocationHistory.find_by nomis_offender_id: prisoner_id_from_url
+    @allocation = AllocationHistory.find_by prison: @prison.code, nomis_offender_id: prisoner_id_from_url
     previous_pom_ids = if @allocation
                          @allocation.previously_allocated_poms
                        else
                          []
                        end
     poms = @prison.get_list_of_poms.index_by(&:staff_id)
     @previous_poms = previous_pom_ids.map { |staff_id| poms[staff_id] }.compact
-    @current_pom = @prison.get_single_pom(@allocation.primary_pom_nomis_id) if @allocation&.primary_pom_nomis_id
+    @current_pom = poms[@allocation.primary_pom_nomis_id] if @allocation&.primary_pom_nomis_id
```

Comment:
I am just trying to make sure I understand this. `@allocation&.primary_pom_nomis_id` checks if the allocation has a primary pom and if it does `poms[@allocation.primary_pom_nomis]` returns the current pom. But if it doesn't have a primary_pom_nomis_id when current_pom get assigned nil?